lib/miq_automation_engine/engine/miq_ae_service_model_base.rb

```diff
@@ -27,11 +27,17 @@ def self.respond_to?(m, *args)
       super
     end
 
+    def self.allowed_find_method?(m)
+      return false if m.starts_with?('find_or_create') || m.starts_with?('find_or_initialize')
+      m.starts_with?('find')
+    end
```

Comment:
@mkanoor We should modify `respond_to?` to use `respond_to_missing?` which you can read about here: https://robots.thoughtbot.com/always-define-respond-to-missing-when-overriding.

If you do that you can remove the `if class_method_exposed?(m)` check in `method_missing`.